_posts/2019-01-04-nextjs.md

```diff
@@ -0,0 +1,290 @@
+};
+```
+
+And we will use that file in our pages/index.js:
+```
+import hotel from '../src/hotel';
+
+export default () => {
+    return (<div>
+        {hotel.name}
+    </div>)
+}
+```
+
+Check in your browser: It now shows us our hotel's name.
+
+## Using another route and a button
+
+What if we want to initially NOT show the data - we only want to reveal it after a click? Well that's easy and we could solve it by adding another page to show the hotel data and link it form the index.js page:
+
+pages/hotel.js
+```
+import hotel from '../src/hotel';
+
+export default () => {
+    return (<div>
+        {hotel.name}
+    </div>)
+}
+```
+
+pages/index.js:
+```
+import hotel from '../src/hotel';
+import Router from 'next/router'
+
+export default () => {
+    return (<div>
+        <button onClick={() => Router.push('/hotel')}>Show Hotel</button>
+    </div>)
+}
+```
+
+Refresh your browser and click the button to go to the /hotel page. Again this looks like very little - but there is a bunch of stuff we get for free.
+
+- we see our routing in action now. And additionally to [links](https://github.com/zeit/next.js/#with-link) we can also imperatively use Router.push
+- if you open your webdeveloper toolbar's network tab you'll see that this uses client side navigation - there is no full reload of our application
+- and again in the webdeveloper toolbar you'll see that we get "bundle splitting" for free: Each page has its own JavaScript bundle - resulting in awesomely fast pages.
+
+Sure it is possible to achieve this with webpack or other bundlers manually as well - but with nextjs it is 0 effort.
+
+## NOT using an additional page
+
+Now lets suppose we have a new requirement: we don't want to have a new url for displaying the hotel. Instead we want a button on the index page that reveals the content - without any url changes.
+
+We can use react and `setState` to achieve this:
+
+```
+import React from 'react';
+import hotel from '../src/hotel';
+
+export default class extends React.Component {
+    state = {
+        showHotel: false
+    }
+    showHotel() {
+        this.setState({ showHotel: true})
+    }
+
+    render() {
+        if (this.state.showHotel) {
+            return <div>{hotel.name}</div>
+        }
+        return (<div>
+            <button onClick={() => this.showHotel()}>Show Hotel</button>
+        </div>)
+    }
+}
+```
+
+But this has one downside: Regardless if the user really looked at the hotel data - he has to load it in his browser. And our PO doesn't like that because it slows down the page for ALL users even when only 1% of them actually click on the "Show Hotel"-button. For our little example that's not much of a difference but in real world applications it sometimes is a huge improvement to only load what's really needed.
+
+We can use dynamic imports to achieve that:
+
+## Dynamic Imports
+
+Next comes with a default webpack configuration that will make dynamic imports just work out of the box.
```

Comment:
You can drop the `just`.

Reply:
I agree that you can usually drop words like `just`, `simply` and so on. But in this case I'm on the edge. The `just` aims to highlight that it does not work out of the box without next.
But I'll just remove it ;)